python-lib/dku_kube/gpu_driver.py

```diff
@@ -23,10 +23,24 @@ def add_gpu_driver_if_needed(cluster_id, kube_config_path, connection_info, tain
     env["KUBECONFIG"] = kube_config_path
 
     # Get the Nvidia driver plugin configuration from the repository
-    nvidia_config_raw = requests.get(
-        "https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/main/deployments/static/nvidia-device-plugin.yml"
-    ).text
-    nvidia_config = yaml.safe_load(nvidia_config_raw)
+    nvidia_config_response = requests.get(
+        "https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/main/deployments/static/nvidia-device-plugin.yml",
+        headers={"User-Agent": "DSS EKS Plugin"}
+    )
```

Comment:
was it intended to remove `.text` here?

Reply:
yes, we keep the response object to ensure its status below and then get the text from the response body if it's successful